calyx-py/calyx/queue_call.py

```diff
@@ -222,13 +223,15 @@ def insert_main(
                 )
             ),
             # If the dataplane component has an answer,
-            # write it to the answer-list and increment the index `j`.
+            # write it to the answer-list and increment the index `i`.
             cb.if_(
                 has_ans.out,
-                [
-                    main.mem_store_d1(ans_mem, j.out, dataplane_ans.out, "write_ans"),
-                    main.incr(j),
-                ],
+                
+                    main.mem_store_d1(ans_mem, i.out, dataplane_ans.out, "write_ans"),
+                    cb.if_(dataplane_err.out, 
+                        main.mem_store_d1(ans_mem, i.out, cb.const(32, 20000), "write_err"), # there was an error
+                        main.mem_store_d1(ans_mem, i.out, cb.const(32, 10000), "write_push"),) # the command was a push
```

Comment:
Nice! But please document these new magic numbers up top in the docstring as well!

Reply:
Also, please explain the logic a little more in these inline comments. I get what you have done:
```python
if has_ans:
  put ans into ans mem    
else: 
  if is_err:
    put error code into ans mem
  else:
    // if we're here, we must be here because we were a successful push.
    put push code into ans mem
```
But I think we'll struggle to understand this from a quick glance at the code in a few weeks time!